src/safeposix/vmmap.rs

```diff
@@ -204,6 +207,18 @@ impl Vmmap {
         npages & !(pages_per_map - 1)
     }
 
+    pub fn set_base_address(&mut self, base_address: i64) {
+        self.base_address = Some(base_address);
+    }
+
+    pub fn user_to_sys(&self, address: i32) -> i64 {
```

Comment:
these functions seem like theyd be better suited for the mem.rs file

Reply:
So do we still store the base_address in vmmap, or we store it in some other place like cage struct?

Reply:
We use the base address on every syscall for translating arguments though, but we pass it every time. I feel like we can unify this into something simpler.

Reply:
Well not every syscall, but something like 50% of them.

Reply:
but we only need base address in dispatcher phase? We shouldn't need base addr in actual syscalls...?

Reply:
per our conversation today were going to keep the base address stored in the vmmap